src/main/java/org/zalando/fahrschein/NakadiReader.java

```diff
@@ -194,23 +193,26 @@ private Cursor readCursor(JsonParser jsonParser) throws IOException {
         final Iterator<T> eventIterator = eventReader.readValues(jsonParser, eventClass);
 
         final List<T> events = new ArrayList<>();
-        while (eventIterator.hasNext()) {
-            readEvent(eventIterator, events);
-        }
-        return events;
-    }
-
-    private void readEvent(final Iterator<T> source, final List<T> target) throws JsonMappingException {
-        try {
-            target.add(source.next());
-        } catch (final RuntimeJsonMappingException e) {
-            final Throwable cause = e.getCause();
-            if (cause instanceof JsonMappingException) {
-                listener.onMappingException((JsonMappingException) cause);
-            } else {
-                throw e;
+        while (true) {
+            try {
+                // MappingIterator#hasNext can theoretically also throw RuntimeExceptions, that's why we use this strange loop structure
```

Comment:
@lukasniemeier-zalando This can happen if the connection breaks (or is just dropped by Nakadi). In this case we can easily recover by reconnecting. In the catch block he explicitly looks for an IOException, so I think it's fine.

We encounter this exact case in our Team an currently recover by catching this IOException and reconnect.